shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -287,6 +292,7 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
 - (void)setViewController:(FlutterViewController*)controller {
   _viewController = controller;
   _mainOpenGLContext = controller.flutterView.openGLContext;
+  _renderTargetPool = [[FlutterRenderTargetPool alloc] initWithGLContext:_mainOpenGLContext];
```

Comment:
It seems like it would be a mistake to create this if the condition on the next line is true.